desktop/core/src/desktop/js/parse/sql/generic/jison/sql_valueExpression.jison

```diff
@@ -919,3 +919,31 @@ CaseWhenThenListPartTwo_EDIT
      $$ = { caseTypes: [{ types: ['T'] }] };
    }
  ;
+
+// ------------------  :: CASTS  --------------------
+
+PrimitiveTypeOrError
+ : PrimitiveType
+ | error
+ ;
+
+ValueExpression
+ : ValueExpression '::' PrimitiveType
+  {
+    parser.extractExpressionText($$, $1, $2, $3);
+    $$ = { types: [ $3.toUpperCase() ] }
+  }
+ ;
+
+ValueExpression_EDIT
+ : ValueExpression '::' AnyCursor
+  {
+    parser.suggestKeywords(parser.getTypeKeywords());
+    $$ = { types: [ 'T' ] };
+  }
```

Comment:
```
AnyCursor
 : 'CURSOR'
 | 'PARTIAL_CURSOR'
 ;
```

The above is the definition of `AnyCursor`, I'm not really sure when to use it vs regular `'CURSOR'` vs `'PARTIAL_CURSOR'`.